Hdcp branch

saik-intel opened the pull request:
**HDCP Daemon integrating with IAS Compositor**
- HDCP daemon will load IAS WL library from the LD_LIBRARY_PATH setting and it will get init/deinit symbols from it and hdcp daemon will communicate with the IAS
-- Added ENV check to identify whether hdcpd running independently or
with IAS to get DRM Master privileges accordingly

satyeshwar committed:
daemon: Changed Status to status

Status is a keyword in several header files and this
causes the daemon to not compile so just changed from upper to
lower case.

Signed-off-by: Satyeshwar Singh <satyeshwar.singh@intel.com>

satyeshwar committed:
daemon: Set content protection through IAS compositor

This change allows the daemon to talk with IAS Wayland compositor
to set the content protection property. It does so by dynamically
loading a library called wl_base and then making the
set_content_protection call into this library.
Note that this library must be present in the same directory as the
daemon. wl_base library is built with the IAS compositor and can be
found here: https://github.com/intel/ias

Signed-off-by: Satyeshwar Singh <satyeshwar.singh@intel.com>

saik-intel committed:
adding libwl_base.so which is library to communic
ate with IAS Wayalnd
- installing this library to where daemon installed

Signed-off-by: kondasai <saikishore.konda@intel.com>

saik-intel committed:
Changes for passing IAS WL library name to HDCP and
extract init/deinit symbols from library
- removing find_libs function

Signed-off-by: kondasai <saikishore.konda@intel.com>

saik-intel committed:
Using env variable to determine hdcp daemon is running with IAS or not

Signed-off-by: kondasai <saikishore.konda@intel.com>

satyeshwar commented on daemon/display_window_util_wl.cpp:
```diff
@@ -23,6 +23,9 @@
 #include <stdlib.h>
 
+/* IAS Wayland library for IPC calls from hdcpd to IAS compositor *
+ * need to set LD_LIBRARY_PATH where this library located */
```
You may want to mention "need to set LD_LIBRARY_PATH only if the library is not in one of the standard paths"

chivakker replied:
This comment shouldn't be needed, unless libwl_base.so has to be placed in particular subdirectory.  Please clarify.

saik-intel replied:
As libwl_base.so is part of /usr/lib64/ias folder  whereas /usr/lib64 is standard library path. I will modify comment

saik-intel replied:
updated the changes

satyeshwar commented on daemon/display_window_util_wl.cpp:
```diff
@@ -99,30 +53,13 @@ bool compare(const class_info *first, const class_info *second)
-			cerr<<"Couldn't open lib"<<(*it)->name<<endl;
-			return 0;
-		}
+        lib_hdl = dlopen(IAS_WL_LIBNAME, RTLD_LAZY | RTLD_GLOBAL);
```
fix your indentation here so that this line is properly indented with other lines.

satyeshwar commented on daemon/display_window_util_wl.cpp:
```diff
@@ -99,30 +53,13 @@ bool compare(const class_info *first, const class_info *second)
-		}
+        lib_hdl = dlopen(IAS_WL_LIBNAME, RTLD_LAZY | RTLD_GLOBAL);
+	if(!lib_hdl) {
+		cerr<<"Couldn't open lib"<<IAS_WL_LIBNAME<<endl;
```
Add space in between lib and "

satyeshwar commented on daemon/display_window_util_wl.cpp:
```diff
@@ -99,30 +53,13 @@ bool compare(const class_info *first, const class_info *second)
+        }
 
 		/* Find init symbol */
 		init = dlsym(lib_hdl, "init");
```
Are all of these lines still indented correctly?

satyeshwar commented on CMakeLists.txt:
```diff
@@ -98,6 +98,8 @@ endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
 install(PROGRAMS ${CMAKE_BINARY_DIR}/hdcpd
         DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT cp)
+install(FILES ${CMAKE_SOURCE_DIR}/lib/libwl_base.so
```
Is your commit message for this commit split between multiple lines? because that's how it looks here. 
Also, Spell Wayland correctly in your commit message.
Add "is" in between "where daemon" and "installed"

chivakker commented on daemon/display_window_util_wl.cpp:
```diff
@@ -0,0 +1,124 @@
+		}
+
+		/* Call class's function */
+		ret = gwl->init();
```
if init fails, can add_reg be called or should it check for ret?

chivakker commented on daemon/display_window_util_wl.cpp:
```diff
@@ -0,0 +1,124 @@
+	/* Find deinit symbol */
+	deinit = dlsym(lib_hdl, "deinit");
+	if(!deinit) {
+		cerr<<"Couldn't open symbol deinit"<<endl;
```
Couldn't find symbol deinit and maybe print dlerror?

saik-intel committed:
Applying the review comments
- Adding dlerror print in case symbol load error
- fixing indentation
- adding init return check to see error and return

Signed-off-by: kondasai <saikishore.konda@intel.com>

shiqiangyu commented on daemon/global_wayland.h:
```diff
@@ -0,0 +1,97 @@
+/*
+** Copyright 2017 Intel Corporation
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
```
One major concern here. Why include so many function prototypes which are not used in global_wayland.h? like registry_handle_global_remove

saik-intel replied:
I guess this file is copied as is from IAS location. @satyeshwar  Can comment on this ?

shiqiangyu commented on daemon/portmanager.cpp:
```diff
@@ -1200,6 +1204,11 @@ int32_t PortManager::SetPortProperty(
+}
+else
+{
+    util_set_content_protection(drmId, *value);
```
util_set_content_protection should have return value for enablement status, the caller of SetPortProperty need this info.

saik-intel replied:
Actually wayland (IAS) set_content_protection is void call thats why we are calling without checking return value as it is void call
@satyeshwar  please comment

shiqiangyu replied:
So if IAS set_content_protection is a void call and need another call for synchronization to get the enablement status, I think we should add one in the wrapper function util_set_content_protection.

saik-intel replied:
With latest commit, we are now supporting the enablement status

shiqiangyu commented on daemon/portmanager.cpp:
```diff
@@ -1147,12 +1148,15 @@ int32_t PortManager::SetPortProperty(
     }
-    
+
+if(ias_env) {
```
indention for this if condition branch.

shiqiangyu commented on daemon/display_window_util_wl.cpp:
```diff
@@ -0,0 +1,130 @@
+typedef void (*deinit_fn)(void *);
+
+
+bool compare(const class_info *first, const class_info *second)
```
seems function compare is not used, please remove it.

saik-intel committed:
- Removing compare function and applying vpg review comments

Signed-off-by: kondasai <saikishore.konda@intel.com>

satyeshwar commented:
That’s not how Wayland works. Wayland calls are asynchronous in nature. An application doesn’t wait for a return value to be sent back by the compositor. If you really want to know whether this call succeeded or not, we will have to make another call into the compositor to get the current content protection status.
-Satyeshwar

From: saik-intel [mailto:notifications@github.com]
Sent: Wednesday, January 23, 2019 1:57 AM
To: intel/hdcp <hdcp@noreply.github.com>
Cc: Singh, Satyeshwar <satyeshwar.singh@intel.com>; Mention <mention@noreply.github.com>
Subject: Re: [intel/hdcp] Hdcp branch (#14)


@saik-intel commented on this pull request.

________________________________

In daemon/portmanager.cpp<https://github.com/intel/hdcp/pull/14#discussion_r250123122>:

> @@ -1200,6 +1204,11 @@ int32_t PortManager::SetPortProperty(

         HDCP_ASSERTMESSAGE("Could not drop drm master privilege");

         return EBUSY;

     }

+}

+else

+{

+    util_set_content_protection(drmId, *value);

Actually wayland (IAS) set_content_protection is void call thats why it is calling without return value.
@satyeshwar<https://github.com/satyeshwar> please comment

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/intel/hdcp/pull/14#discussion_r250123122>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAy3W5gFsvAMgNlkz9Qqsj4n7g_FmNqFks5vGDHegaJpZM4aHnBu>.

saik-intel committed:
Now Getting hdcp status success or failure from util_set_content_protection  via IAS compositor

Signed-off-by: kondasai <saikishore.konda@intel.com>

saik-intel committed:
Now Getting hdcp status success of failue from util_set_content_protection  via IAS compositor

Signed-off-by: kondasai <saikishore.konda@intel.com>

saik-intel committed:
Merge branch 'hdcp-branch' of https://github.com/saik-intel/hdcp into hdcp-branch

saik-intel committed:
removing not required functions used by hdcp module from global_wayland header file

Signed-off-by: kondasai <saikishore.konda@intel.com>

saik-intel commented:
Now all changes requested by reviewers are done. Please review and merge to master branch
